src/main/java/com/example/onlinebookshop/services/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,175 @@
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderDto> getOrders(Pageable pageable) {
+        return orderRepository.findAll().stream()
+                .map(orderMapper::orderToOrderDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderDto> getOrdersByUserId(Long userId, Pageable pageable) {
+        return orderRepository.findAllByUserId(userId, pageable).stream()
+                .map(orderMapper::orderToOrderDto)
+                .toList();
+    }
+
+    @Transactional
+    @Override
+    public OrderDto addOrder(Long userId, CreateOrderDto createOrderDto) {
+        if (createOrderDto.shippingAddress().isBlank()) {
+            throw new AddressNotFoundException(
+                    "Can't form an order with no address. Try again and be sure to input it.");
+        }
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId);
+        if (shoppingCart == null) {
+            throw new EntityNotFoundException("Can't find shopping cart for user " + userId);
+        }
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new OrderProcessingException(
+                    "To post an order, you should have at least 1 product in your cart");
+        }
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setStatus(Status.CREATED);
+        setOrderItems(shoppingCart, order);
+        order.setTotal(order.getOrderItems().stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add)
+        );
+        order.setShippingAddress(createOrderDto.shippingAddress());
+        setOrderTime(order);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(order.getOrderItems());
+        shoppingCart.getCartItems().clear();
+        shoppingCartRepository.save(shoppingCart);
+        return orderMapper.orderToOrderDto(order);
+    }
+
+    @Override
+    public OrderDto updateOrderStatus(Long orderId, UpdateOrderDto updateOrderDto) {
+        Order order = orderRepository.findById(orderId)
+                .map(o -> {
+                    o.setStatus(Status.valueOf(updateOrderDto.status()
+                            .toUpperCase()
+                            .trim()));
+                    return o;
+                })
+                .orElseThrow(() -> new EntityNotFoundException(
+                        String.format("Order with id: %d not found", orderId)
+                ));
+        orderRepository.save(order);
+        return orderMapper.orderToOrderDto(order);
+    }
+
+    @Override
+    public List<OrderItemDto> findOrderItemsByOrderId(Long orderId) {
+        Optional<Order> optionalOrder = orderRepository.findById(orderId);
```

Comment:
Can we use the same approach like in below method?
```suggestion
        Order order = orderRepository.findById(orderId).orElseThrow(() -> 
        return orderItemMapper.toOrderItemDtoList(order.getOrderItems());
```